ui/panels-plugin/src/plugins/line-chart/LineChartContainer.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useMemo } from 'react';
+import { GridComponentOption, ToolboxComponentOption } from 'echarts';
+import { Box, Skeleton } from '@mui/material';
+import { LineChart, EChartsDataFormat, EChartsValues } from '@perses-dev/components';
+import { useRunningGraphQueries } from './GraphQueryRunner';
+import { getRandomColor } from './utils/palette-gen';
+import { getCommonTimeScale, getXValues } from './utils/data-transform';
+
+export const OPTIMIZED_MODE_SERIES_LIMIT = 500;
+
+export const EMPTY_GRAPH_DATA = {
+  timeSeries: [],
+  xAxis: [],
+};
+
+export interface LineChartContainerProps {
+  width: number;
+  height: number;
+}
+
+/**
+ * Draws a LineChart with Apache ECharts for the current running time series.
+ */
+export function LineChartContainer(props: LineChartContainerProps) {
```

Comment:
Thoughts on a better name for this? I tried combining this component with LineChartPanel.tsx, but keeping them separate was easier (open to any ideas here)

Reply:
(discussed off-line with Luke) Reason is GraphQueryRunner passes data via context, but the chart needs it as a prop. Short term we're going to leave LineChartContainer as is, until graph queries approach is potentially updated to take advantage of react-query's `useQueries` hook.